core/bench_contract_test.go

```diff
@@ -0,0 +1,177 @@
+	receipt.Bloom = types.CreateBloom(types.Receipts{receipt})
+	receipt.BlockHash = blockHash
+	receipt.BlockNumber = blockNumber
+	receipt.TransactionIndex = uint(statedb.TxIndex())
+	return receipt, result, err
+}
+
+func ApplyTransactionAndGetResult(config *params.ChainConfig, bc ChainContext, blockContext vm.BlockContext, gp *GasPool, statedb *state.StateDB, header *types.Header, tx *types.Transaction, usedGas *uint64, cfg vm.Config) (*types.Receipt, *ExecutionResult, error) {
+	msg, err := TransactionToMessage(tx, types.MakeSigner(config, header.Number, header.Time), header.BaseFee)
+	if err != nil {
+		return nil, nil, err
+	}
+	// Create a new context to be used in the EVM environment
+	vmenv := vm.NewEVM(blockContext, vm.TxContext{}, statedb, config, cfg)
+	return applyTransactionAndGetResult(msg, config, gp, statedb, header.Number, header.Hash(), tx, usedGas, vmenv)
+}
+
+func (b *BlockGen) AddTxOrFail(tx *types.Transaction) (*types.Receipt, error) {
+	if b.gasPool == nil {
+		b.SetCoinbase(common.Address{})
+	}
+	b.statedb.SetTxContext(tx.Hash(), len(b.txs))
+	blockContext := NewEVMBlockContext(b.header, nil, &b.header.Coinbase)
+	receipt, result, err := ApplyTransactionAndGetResult(b.config, nil, blockContext, b.gasPool, b.statedb, b.header, tx, &b.header.GasUsed, vm.Config{})
+	if err != nil {
+		return nil, err
+	}
+
+	b.txs = append(b.txs, tx)
+	b.receipts = append(b.receipts, receipt)
+
+	if result.Err != nil {
+		return receipt, result.Err
+	}
+	return receipt, nil
+}
+
+func stressTestTrieDb(numContracts int, callsPerBlock int, elements int64, gasTxLimit uint64) func(int, *BlockGen) {
+	contractAddr := make([]common.Address, numContracts)
+	contractTxs := make([]*types.Transaction, numContracts)
+
+	gasPrice := big.NewInt(225000000000)
+	gasCreation := uint64(258000)
+	deployedContracts := 0
+
+	for i := 0; i < numContracts; i++ {
+		nonce := uint64(i)
+		tx, _ := types.SignTx(types.NewContractCreation(nonce, big.NewInt(0), gasCreation, gasPrice, common.FromHex(stressBinStr)), signer, testKey)
+		sender, _ := types.Sender(signer, tx)
+		contractTxs[i] = tx
+		contractAddr[i] = crypto.CreateAddress(sender, nonce)
+	}
+
+	stressABI := contract.ParseABI(stressABIStr)
+	txPayload, _ := stressABI.Pack(
+		"writeValues",
+		big.NewInt(elements),
+	)
+
+	return func(i int, gen *BlockGen) {
+		if len(contractTxs) != deployedContracts {
+			block := gen.PrevBlock(i - 1)
+			gas := block.GasLimit()
+			for ; deployedContracts < len(contractTxs) && gasCreation < gas; deployedContracts++ {
+				if receipt, err := gen.AddTxOrFail(contractTxs[deployedContracts]); err != nil {
+					fmt.Printf("\nBlock: %d\nReceipt: %+v\n", i, receipt)
+					panic(err)
+				}
+				gas -= gasCreation
+			}
+			return
+		}
+
+		for e := 0; e < callsPerBlock; e++ {
+			tx, err := types.SignTx(types.NewTransaction(gen.TxNonce(benchRootAddr), contractAddr[i%deployedContracts], big.NewInt(0), gasTxLimit, gasPrice, txPayload), signer, testKey)
```

Comment:
q: why we use `benchRootAddr` for nonce?

Reply:
@ceyonur I think benchRootAddr is the one signing these transactions. Most of this is boilerplate from other benchmarks. I can make another's keys to sign transactions, it would make no difference imho.

Reply:
We should use `benchRootKey`. Luckily these two were defined for same addresses, otherwise this would not work: 
https://github.com/ava-labs/subnet-evm/blob/trie-stress-test/core/bench_test.go#L88
https://github.com/ava-labs/subnet-evm/blob/trie-stress-test/core/state_processor_test.go#L51